### DISCUSSION TITLE
Compatibility with .NET Framework (6.0,6.1,6.2)

### DISCUSSION THREAD

**Discussion by CrhistopherDiaz** [jchristn/WatsonTcp#257] [Q&A]
good evening and sorry for the inconvenience right now I am using net framework 6.2 which is a net framework compatible with all versions of Windows now there is a problem watson tcp connects to the server without problems but when sending a message it does not send it now yes I install net framework 7 it works without problem could you correct this error

**Comment by Stupeflip** [comment]
The library that you are trying to use seems to be using the Tcp class from the .NET (not .NET Framework) System.Net.Sockets library. This library is not compatible with the outdated .NET Framework versions such as 6.0, 6.1, and 6.2.

This means that if you are using .NET Framework 6.0, 6.1, or 6.2, you won't be able to use this library as it is built using code that is not compatible with those versions. You may need to either update to a newer version of .NET Framework that supports the updated Tcp class or look for a different library that is compatible with your version of .NET Framework.
